src/main/java/org/assertj/core/api/AbstractCallableAssert.java

```diff
@@ -0,0 +1,60 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ *
+ * Copyright 2012-2022 the original author or authors.
+ */
+package org.assertj.core.api;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+import java.util.concurrent.Callable;
+
+import org.assertj.core.util.CheckReturnValue;
+
+public abstract class AbstractCallableAssert<SELF extends AbstractCallableAssert<SELF, V>, V>
+    extends AbstractThrowingExecutableAssert<SELF, Callable<V>> {
+
+  protected V result;
+
+  protected AbstractCallableAssert(Callable<V> actual, Class<?> selfType) {
+    super(actual, selfType);
+  }
+
+  @Override
+  protected void execute(Callable<V> actual) throws Exception {
```

Comment:
is it worth specifying that this throws `Throwable` directly? Shouldn't make much difference but semantically it'd make sense to be able to catch Errors and the likes if testing code that could raise errors directly, which I assume this would do already.

Reply:
Right now, it wouldn't catch `Throwable` or `Error` instances as the input parameter is a `Callable`, and [`Callable::call`](https://docs.oracle.com/en/java/javase/21/docs/api/java.base/java/util/concurrent/Callable.html#call()) allows throwing `Exception`s only.

Maybe we should consider if a new Callable-like interface that throws `Throwable` should be defined instead of relying on the JDK `Callable`.

@ascopes do you already have a concrete use case that would require it?

Reply:
As suggested by @sbrannen in https://github.com/assertj/assertj/issues/3314#issuecomment-1874217792, a `ThrowingSupplier` could be defined if we decide that `Callable` throwing `Exception` is not flexible enough.